ods_tools/oed/source.py

```diff
@@ -219,6 +219,7 @@ def from_dataframe(cls, exposure, oed_type, oed_df: pd.DataFrame):
                 pd_dtype[column] = 'category'
             if pd_dtype[column] == 'category':  # we need to convert to str first
                 to_tmp_dtype[column] = 'str'
+                oed_df.loc[oed_df[column].isin(BLANK_VALUES), column] = ''
```

Comment:
Need to change this line, otherwise for columns with no blank values it tried to assign `''` to an empty Series

```
> column
'PortNumber'

> oed_df['PortNumber']
0    1
1    1
Name: PortNumber, dtype: category
Categories (1, object): ['1']

> oed_df.loc[oed_df[column].isin(BLANK_VALUES), column]
Series([], Name: PortNumber, dtype: category
Categories (1, object): ['1'])

> oed_df.loc[oed_df[column].isin(BLANK_VALUES), column] = ''
*** TypeError: Cannot setitem on a Categorical with a new category (), set the categories first
```

Reply:
CI retest - https://github.com/OasisLMF/OasisPlatform/actions/runs/4469717035